libs/labelbox/src/labelbox/schema/api_key.py

```diff
@@ -0,0 +1,407 @@
+                f"Error retrieving user with email '{email}': {str(e)}"
+            )
+
+    @staticmethod
+    def create_api_key(
+        client: "Client",
+        name: str,
+        user: Union["User", str],
+        role: Union["Role", str],
+        validity: int = 0,
+        time_unit: TimeUnit = TimeUnit.SECOND,
+    ) -> Dict[str, str]:
+        """Creates a new API key using the provided client.
+
+        Args:
+            client: The Labelbox client instance.
+            name (str): Name of the API key
+            user (Union[User, str]): User object or email for whom to create the API key
+            role (Union[Role, str]): Permission role for the API key (Role enum or string)
+            validity (int, optional): Validity period value (must be positive). Defaults to 0.
+            time_unit (TimeUnit, optional): Time unit for validity period. Defaults to TimeUnit.SECOND.
+
+        Returns:
+            Dict[str, str]: Dictionary containing the created API key details including id and jwt
+
+        Raises:
+            ValueError: If invalid parameters are provided
+            LabelboxError: If the API request fails or there are permission issues
+        """
+        if not name or not isinstance(name, str):
+            raise ValueError("name must be a non-empty string")
+
+        user_email = user.email if hasattr(user, "email") else user
+        if not user_email or not isinstance(user_email, str):
+            raise ValueError("user must be a User object or a valid email")
+
+        # Check if the user exists in the organization
+        user_id = ApiKey._get_user(client, user_email)
+        if not user_id:
+            raise ValueError(
+                f"User with email '{user_email}' does not exist in the organization"
+            )
+
+        role_name = role.name if hasattr(role, "name") else role
+        if not role_name or not isinstance(role_name, str):
+            raise ValueError("role must be a Role object or a valid role name")
+
+        allowed_roles = ApiKey._get_available_api_key_roles(client)
+        if role_name not in allowed_roles:
+            raise ValueError(
+                f"Invalid role specified. Allowed roles are: {allowed_roles}"
+            )
+
+        validity_seconds = 0
+        if validity > 0:
+            if not isinstance(time_unit, TimeUnit):
+                raise ValueError(
+                    "time_unit must be a valid TimeUnit enum value"
+                )
+
+            validity_seconds = validity * time_unit.value
+
+            if validity_seconds < TimeUnit.MINUTE.value:
+                raise ValueError("Minimum validity period is 1 minute")
+
+            max_seconds = 25 * TimeUnit.WEEK.value
+            if validity_seconds > max_seconds:
+                raise ValueError(
+                    "Maximum validity period is 6 months (or 25 weeks)"
+                )
+        else:
+            raise ValueError("validity must be a positive integer")
```

Comment:
could just flip logic here and get rid of nested ifs for better code readability. Otherwise, too much mental gymnastic to perform for simple logic

Reply:
We still support 3.9. Otherwise, I would use pattern matching to simplify the structure for the bunch of tests.
I will update the test for "validity".